src/prefect/tasks/prometheus/pushgateway.py

```diff
@@ -0,0 +1,203 @@
+import abc
+from prometheus_client import (
+    CollectorRegistry,
+    Gauge,
+    pushadd_to_gateway,
+    push_to_gateway,
+)
+from typing import Dict, List, Optional
+from prefect import Task
+from prefect.utilities.tasks import defaults_from_attrs
+
+
+class _GaugeToGatewayBase(Task):
+    @abc.abstractmethod
+    def _push(self, **push_args):
+        pass
+
+    def __init__(
+        self,
+        pushgateway_url: str = None,
+        counter_name: str = None,
+        counter_description: str = None,
+        grouping_key: List[str] = None,
+        job_name: str = None,
+        **kwargs,
+    ):
+        self.pushgateway_url = pushgateway_url
+        self.counter_name = counter_name
+        self.counter_description = counter_description
+        self.job_name = job_name
+        self.grouping_key = grouping_key
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs(
+        "pushgateway_url",
+        "counter_name",
+        "counter_description",
+        "job_name",
+        "grouping_key",
+    )
+    def run(
+        self,
+        values: List[float],
+        labels: List[Dict[str, str]],
+        pushgateway_url: str = None,
+        counter_name: str = None,
+        counter_description: str = None,
+        grouping_key: Optional[List[str]] = None,
+        job_name: str = None,
+    ) -> None:
+        """
+        Push a Gauge to prometheus [PushGateway](ttps://prometheus.io/docs/practices/pushing/).
+        This will unpack the values and labels to create all the a collector that will be pushed
+        to pushgateway.
+
+        Args:
+            - values (List[float]): List of the values to push
+            - labels (List[Dict[str, str]]): List of the labels to push attached to the values
```

Comment:
Aren't labels optional? Is there a reason to make this required? https://github.com/prometheus/client_python#gauge

Reply:
yes they are, however you need to match the values. so you would need to set empty dict in this case